### ISSUE TITLE
node_feas introduces look-ahead bias to the predictions?

### ISSUE THREAD

**Issue by turmeric-blend** [chaoshangcs/GTS#2]
Hi, I think since your model uses [`node_feas`](https://github.com/chaoshangcs/GTS/blob/208bc8bb5afbbdd1ea6cddbecbb9ccd0e9effdc7/model/pytorch/model.py#L208) (to build adj matrix) for train/validation/test, it might have introduced look-ahead bias (data leakage) to your predictions?

It is still ok if you use it for training only, but since from the code, it is used during evaluation as well, this might have contributed to the exceptionally good performance? Please point out if I have missed something.

Thank you.

**Comment by chaoshangcs** [comment]
Hi, thanks for your great question. The graph generator aims to estimate matrix Bernoulli distribution. Previously LDS method did it without using any information. Here the training set is like a prior knowledge to help you to learn the matrix Bernoulli distribution. It takes advantage of the relations between features. So it will be fine to use it directly. Absolutely, your suggestion is another way to do experiments. We haven't tried it before. It might be able to improve the performance. If you get any interesting results, look forward to your sharing.

**Comment by turmeric-blend** [comment]
The training dataset (**which contains the entire lookback and horizon sequence length up to max training length**) is used to learn the prior during the training loop. This is all good. But my question is what do you pass as `node_feas` during the validation/test loop? training, validation, or test dataset?

**Comment by chaoshangcs** [comment]
Thanks for your question. It is still the training dataset.

**Comment by turmeric-blend** [comment]
Thank you.
